ext/storage_sources/s3_store/s3_storage_source.cpp

```diff
@@ -405,6 +408,59 @@ S3Terminate(WT_STORAGE_SOURCE *storage, WT_SESSION *session)
     return (0);
 }
 
+/*
+ * S3Flush --
+ *     Flush file to S3 Store using AWS PutObject.
+ */
+static int
+S3Flush(WT_STORAGE_SOURCE *storageSource, WT_SESSION *session, WT_FILE_SYSTEM *fileSystem,
+  const char *source, const char *object, const char *config)
+{
+    S3_FILE_SYSTEM *fs = (S3_FILE_SYSTEM *)fileSystem;
+    int ret = fs->connection->PutObject(fs->bucketName, object, source);
+    return (ret);
+}
+
+/*
+ * S3FlushFinish --
+ *     Flush local file to cache.
+ */
+static int
+S3FlushFinish(WT_STORAGE_SOURCE *storage, WT_SESSION *session, WT_FILE_SYSTEM *fileSystem,
+  const char *source, const char *object, const char *config)
+{
+    S3_STORAGE *s3 = (S3_STORAGE *)storage;
+    std::string srcPath = S3HomePath(fileSystem, source);
+    std::string destPath = S3CachePath(fileSystem, source);
+    std::cout << srcPath << destPath << std::endl;
+    int ret;
+    if ((ret = link(srcPath.c_str(), destPath.c_str())) != 0)
+        return ret;
+    if ((ret = chmod(destPath.c_str(), 0444)) != 0)
+        return ret;
+    return ret;
+}
+
+/*
+ * S3CachePath --
+ *     Construct the cache pathname from the file system and local name.
+ */
+std::string
+S3CachePath(WT_FILE_SYSTEM *fileSystem, const char *name)
+{
+    return S3Path(((S3_FILE_SYSTEM *)fileSystem)->cacheDir, name);
+}
+
+/*
+ * S3HomePath --
+ *     Construct the source pathname from the file system and local name.
+ */
+std::string
```

Comment:
Same as above, consider `GetS3HomePath`. Do we plan to use this method more than once? Maybe it doesn't need to be a method?

Reply:
I currently am only using this function once. Happy to take it out as I do not see any use for it in the future currently!

Reply:
You can take out this one and the one above, I think!

Reply:
Oh, I think you already did :)